scripts/Imaging/imgpy/helper.py

```diff
@@ -0,0 +1,340 @@
+"""
+
+# do we want to wrap this in a global class?
+# so that helper will just store the funcitonality, but helper_data will
+# store the actual data?
+
+_whole_exec_timer = None
+_timer_running = False
+_timer_start = None
+_timer_print_prefix = " ---"
+
+_verbosity = 2
+
+_cache_last_memory = None
+
+_note_str = " > Note: "
+_warning_str = " >> WARNING: "
+_error_str = " >>> ERROR: "
+
+# for timer
+_progress_bar = None
+
+_readme = None
+
+
+def check_config_class(config):
+    from configs.recon_config import ReconstructionConfig
+    assert isinstance(
+        config, ReconstructionConfig
+    ), "The provided config is not of type ReconstructionConfig and cannot be used!"
+
+
+def initialise(config, saver=None):
+    global _verbosity
+    global _readme
+    _verbosity = config.func.verbosity
+    if saver is not None:
+        from readme import Readme
+        _readme = Readme(config, saver)
+
+
+def check_config_integrity(config):
+    check_config_class(config)
+
+    if not config.func.output_path:
+        tomo_print_warning(
+            "No output path specified, no output will be produced!")
+
+
+def check_data_stack(data):
+    import numpy
+
+    if not isinstance(data, numpy.ndarray):
+        raise ValueError(
+            "Invalid stack of images data. It is not a numpy array: {0}".
+            format(data))
+
+    if 3 != len(data.shape):
+        raise ValueError(
+            "Invalid stack of images data. It does not have 3 dimensions. Shape: {0}".
+            format(data.shape))
+
+
+def debug_print_memory_usage_linux(message=""):
+    try:
+        # Windows doesn't seem to have resource package, so this will
+        # silently fail
+        import resource as res
+        print(" >> Memory usage", res.getrusage(res.RUSAGE_SELF).ru_maxrss,
+              "KB, ", int(res.getrusage(res.RUSAGE_SELF).ru_maxrss) / 1024,
+              "MB", message)
+    except ImportError:
+        res = None
+        pass
+
+
+def progress_available():
+    try:
+        from tqdm import tqdm
+        return tqdm
+    except ImportError:
+        tomo_print_note("Progress bar library TQDM not available. "
+                        "To install locally please use pip install tqdm. "
+                        "Falling back to ASCII progress bar.")
+
+
+def run_import_checks(config):
+    """
+    Run the import checks to notify the user which features are available in the execution.
+    """
+    from parallel import utility as pu
+    progress_available()
+
+    if not pu.multiprocessing_available():
+        tomo_print_note("Multiprocessing not available.")
+    else:
+        tomo_print_note(
+            "Running process on {0} cores.".format(config.func.cores))
+
+
+def get_memory_usage_linux():
+    try:
+        # Windows doesn't seem to have resource package, so this will
+        # silently fail
+        import resource as res
```

Comment:
If `res = None` isn't present in the except part, then pep8 will complain for not being declared in both. Also why move the print? If I move the print outside it'll start printing nonsense if the function is left by accident in the code